src/main/java/ca/gc/cyber/ops/assemblyline/java/client/AssemblylineClientConfig.java

```diff
@@ -2,42 +2,56 @@
 
 @Configuration
 @EnableConfigurationProperties(
-        {ApikeyAuthProperties.class, PasswordAuthProperties.class, AssemblylineClientProperties.class,
-                ProxyProperties.class})
+        {ApikeyAuthProperties.class, PasswordAuthProperties.class, AssemblylineClientProperties.class})
+@Slf4j
 public class AssemblylineClientConfig {
 
+    @Bean
+    @ConditionalOnProperty("assemblyline-java-client.url")
+    public AssemblylineClient assemblylineClient(HttpClient httpClient, AssemblylineAuthenticationMethod authMethod,
+                                                 ObjectMapper defaultMapper,
+                                                 AssemblylineClientProperties assemblylineClientProperties) {
+        return new AssemblylineClient(assemblylineClientProperties, httpClient, defaultMapper,
+                authMethod);
+    }
+
     @Bean
     @ConditionalOnProperty(name = "assemblyline-java-client.auth-method", havingValue = "apikey")
-    public AssemblylineClient assemblylineClientApiAuth(ApikeyAuthProperties apikeyAuthProperties,
-                                                        ObjectMapper defaultMapper,
-                                                        AssemblylineClientProperties assemblylineClientProperties,
-                                                        ProxyProperties proxyProperties) {
-        ApikeyAuthentication apikeyAuthentication = new ApikeyAuthentication(apikeyAuthProperties);
-        return new AssemblylineClient(assemblylineClientProperties, proxyProperties, defaultMapper,
-                apikeyAuthentication);
+    public AssemblylineAuthenticationMethod assemblylineClientApiAuth(ApikeyAuthProperties apikeyAuthProperties) {
+        return new ApikeyAuthentication(apikeyAuthProperties);
     }
 
     @Bean
     @ConditionalOnProperty(name = "assemblyline-java-client.auth-method", havingValue = "password")
-    public AssemblylineClient assemblyLineClientPasswordAuth(PasswordAuthProperties passwordAuthProperties,
-                                                             ObjectMapper defaultMapper,
-                                                             AssemblylineClientProperties assemblyLineClientProperties,
-                                                             ProxyProperties proxyProperties) {
-        PasswordAuthentication passwordAuthentication = new PasswordAuthentication(passwordAuthProperties);
-        return new AssemblylineClient(assemblyLineClientProperties, proxyProperties, defaultMapper,
-                passwordAuthentication);
+    public AssemblylineAuthenticationMethod assemblyLineClientPasswordAuth(PasswordAuthProperties passwordAuthProperties) {
+        return new PasswordAuthentication(passwordAuthProperties);
+    }
+
+    /**
+     * Returns a basic HttpClient with HTTPS support enabled.
+     *
+     * @return HttpClient
+     */
+    @Bean
+    @ConditionalOnMissingBean
+    public HttpClient httpClient() {
```

Comment:
I like delegating to the client the creation of the HttpClient, since this gives the client more flexibility. I am not convinced this conditional is a good approach here. An application could define multiple HttpClient beans, and maybe none of them is for AL.

Reply:
Hmm. I could annotate this bean and the parameter on the client creation method with something like `@Qualifier("assemblylineClientHttpClient")`, but then that would _require_ users to annotate their `HttpClient` bean the same way, which loses some of the "magic" of just picking up an existing bean.

Another possible option would be to annotate the `assemblylineClient()` method as `@ConditionalOnMisingBean` so that users can override the creation and use whatever `HttpClient` they want, with whatever qualifiers they want, while keeping the "magic" for the simple use cases.

Reply:
I'm good with any option. It just needs to be documented.

Reply:
Does this need to be a bean? If AssemblyLineClient is the bean the client will customize, the client will likely create its own HttpClient bean.

Also, if the client creates its own AssemblyLineClient with its own HttpClient that may not be named httpClient, this bean will sit here unused?

I just don't see the benefit of making it a bean. Maybe I am missing something?

Reply:
Ignore my comment above. For just setting proxy stuff like in your example, just the bean need to change.

Reply:
I still wonder if the bean should at least be named different, like assmblyLineHttpClient, so the client application can have its own httpClient that won't conflict.

Reply:
I've changed the name to `assemblylineHttpClient`.